dotcom-rendering/src/web/components/ServerSideOnly.tsx

```diff
@@ -0,0 +1,27 @@
+type Props = {
+	children: React.ReactNode;
+};
+
+/**
+ * ServerSideOnly - Use this if you have some JSX you want to be rendered on the server - but not hydrated on the client
+ *
+ * How does this work? When on the server, this component will render its children within a <div> normally.
+ * 	Once on the client, it will render a <div> with an empty dangerouslySetInnerHTML. This communicates to (p)react that there
+ * 	should be HTML in the <div>, but because it's set with dangerouslySetInnerHTML, (p)react will not parse it as part of hydration.
+ *
+ * When do I need this? DCR will only hyrdrate code that lives within islands -
+ * 	you should only need to use this component if you have specific code within an island you do not want to be hydrated.
+ *
+ * Note: Keep in mind this function prevents hydration, but does not prevent code from being included in the webpack bundle.
```

Comment:
Wow :)

Normally when hydrating, the SSR output and the hydrated component [have to match](https://reactjs.org/docs/react-dom.html#hydrate) in order for hydration to correctly 'mount'.

So just for clarification is the `dangerouslySetInnerHTML` a P/React 'trick' to skip hydration for that particular component and therefore the hydration rule around matching output is locally suspended and does not apply?